src/pages/index.js

```diff
@@ -56,31 +73,81 @@ function handleImageClick(data) {
 }
 
 function handleEditProfileFormSubmit(inputValues) {
-  userInfo.setUserInfo({
-    name: inputValues.title,
-    job: inputValues.description,
-  });
+  const { title, description } = inputValues;
+  api
+    .updateProfile(title, description)
+    .then((data) => {
+      userInfo.setUserInfo({
+        name: data.name,
+        job: data.about,
+      });
+      document.querySelector("#profile-name").textContent = data.name;
+      document.querySelector("#profile-description").textContent = data.about;
+    })
+    .catch((err) => console.error(`Error updating profile:", ${err}`));
   editProfileModal.close();
 }
 
+function handleDeleteCard(card) {
+  {
+    confirmPopup.setSubmitFunction(() => {
+      api
+        .deleteCard(card._id)
+        .then(() => {
+          card.remove();
+        })
+        .catch((err) => {
+          console.error(`Error deleting card: ${err}`);
+        });
+      confirmPopup.close();
+    });
+
+    confirmPopup.open();
+  }
+}
+
 function handleAddCardFormSubmit(inputValues) {
+  addCardModal.setSaving(true);
   const cardData = { name: inputValues.title, link: inputValues.url };
-  const newCard = createCard(cardData);
-  cardSection.addItem(newCard);
-  addCardModal.close();
-  modalAddForm.reset(); // clear the form
-  addCardFormValidator.disableButton(); // disable the button
+  api
+    .addCard(cardData)
+    .then((newCard) => {
+      const cardElement = createCard(newCard);
+      cardSection.addItem(cardElement);
+      addCardModal.close();
+      modalAddForm.reset(); // clear the form
+      addCardFormValidator.disableButton(); // disable the button
+    })
+    .finally(() => addCardModal.setSaving(false));
 }
 
-function createCard({ name, link }) {
-  const card = new Card({ name, link }, "#card-template", handleImageClick);
+function createCard(data) {
+  const card = new Card(
+    data,
+    "#card-template",
+    handleImageClick,
+    handleDeleteCard,
+    handleLikeClick
+  );
   const cardElement = card.getView();
   return cardElement;
 }
 
+function handleProfileImageFormSubmit(inputValues) {
+  const imageData = { avatar: inputValues.avatar };
+  api.updateProfilePicture(imageData).then((data) => {
+    const profileImage = document.querySelector(".profile__image");
+    profileImage.src = data.avatar;
+    profileImageModal.close();
+  });
+  profileImageFormValidator.disableButton(config);
+  profileImageForm.reset();

```

Comment:
These 2 lines should be in a `then` block

You need to **disable** the button only in a `then` block after a **successful response** to be able to click it again if there is a server error.